isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java

```diff
@@ -100,6 +107,57 @@ public RelNode visit(Project project) throws RuntimeException {
     return relBuilder.push(child).project(rexList).build();
   }
 
+  @Override
+  public RelNode visit(Sort sort) throws RuntimeException {
+    RelNode child = sort.getInput().accept(this);
+    List<RelFieldCollation> relFieldCollations =
+        sort.getSortFields().stream().map(sortField -> toRelFieldCollation(sortField)).toList();
+    if (relFieldCollations.isEmpty()) {
+      return relBuilder.push(child).sort(Collections.EMPTY_LIST).build();
+    }
+    return relBuilder.push(child).sort(RelCollations.of(relFieldCollations)).build();
+  }
+
+  @Override
+  public RelNode visit(Fetch fetch) throws RuntimeException {
+    RelNode child = fetch.getInput().accept(this);
+    var optCount = fetch.getCount();
+    long count = -1L;
+    if (optCount.isPresent()) {
+      count = optCount.getAsLong();
+    }
+    var offset = fetch.getOffset();
+    return relBuilder.push(child).limit((int) offset, (int) count).build();
```

Comment:
Let's put a check above this that throws if these truncations are going to cause a lost of value (as opposed to silently truncating).

Reply:
done